### PR TITLE
Allow webdev to initiate main

### PR DESCRIPTION
**Pull request by grouma**
- Add new webdev request `RunMain`
- Add configuration to auto run main
- Update the daemon command to only run main after all listeners have been set up

Closes https://github.com/dart-lang/webdev/issues/251

### CODE REVIEW

File: webdev/build.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ targets:
         options:
           compiler: dart2js
           dart2js_args:
-            - -O4
+            - -O0
```
Comment (jakemac53) [inline webdev/build.yaml]: remember to revert this before submitting

File: webdev/lib/src/serve/middlewares/injected_middleware.dart [hunk 2]
```diff
@@ -56,6 +56,9 @@ Handler Function(Handler) createInjectedHandler(
             var requestedUri = request.requestedUri;
             var appId =
                 base64.encode(md5.convert(utf8.encode('$requestedUri')).bytes);
+            // We prevent main from being called and make it runnable through
+            // a global variable.
+            body = body.replaceFirst('app.main.main();', '');
```
Comment (jakemac53) [inline webdev/lib/src/serve/middlewares/injected_middleware.dart]: This is actually a problem - the first `main` part here I believe is derived from the dart file name so this won't work if its called something other than `main`.

Comment (grouma) [reply webdev/lib/src/serve/middlewares/injected_middleware.dart]: Oh fun. I'll add a test and let you solve this. Alternatively we know it's the next line after the injection string. May be less flaky than a regex.

Comment (grouma) [reply webdev/lib/src/serve/middlewares/injected_middleware.dart]: I went ahead and implemented my suggestion. I'm not thrilled with it but it's probably better than a regex? Best solution is to probably make a robust marker in build_web_compilers.

File: example/web/main.dart [hunk 3]
```diff
@@ -7,6 +7,8 @@ import 'dart:developer';
 import 'dart:html';
 
 void main() {
+  print('Initial Print');
```
Comment (kevmoo) [inline example/web/main.dart]: debug left-over?

Comment (grouma) [reply example/web/main.dart]: No. This is required for the test I added. The issue I'm resolving is that we did not capture these prints prior to this PR.

**Pull request merged**